x/storage/keeper/payment_check.go

```diff
@@ -0,0 +1,283 @@
+					Detail{bucket.BucketName, flow.Rate})
+			}
+
+			// user payment account
+			expectedNetFlowRate = expectedNetFlowRate.Neg()
+			_, ok := userFlowRateMap[paymentAddress]
+			if !ok {
+				userFlowRateMap[paymentAddress] = sdkmath.ZeroInt()
+				userFlowRateDetailMap[paymentAddress] = []Detail{}
+			}
+			userFlowRateMap[paymentAddress] = userFlowRateMap[paymentAddress].Add(expectedNetFlowRate)
+			userFlowRateDetailMap[paymentAddress] = append(userFlowRateDetailMap[paymentAddress],
+				Detail{bucket.BucketName, expectedNetFlowRate})
+		}
+
+		// get lock balance
+		objectPrefixStore := prefix.NewStore(store, types.GetObjectKeyOnlyBucketPrefix(bucket.BucketName))
+		it := objectPrefixStore.Iterator(nil, nil)
+		defer it.Close()
+
+		expectedLockBalance := sdkmath.ZeroInt()
+		for ; it.Valid(); it.Next() {
+			u256Seq := sequence.Sequence[sdkmath.Uint]{}
+			objectInfo, found := k.GetObjectInfoById(ctx, u256Seq.DecodeSequence(it.Value()))
+			if found && (objectInfo.ObjectStatus == types.OBJECT_STATUS_CREATED || objectInfo.IsUpdating) {
+				priceTime := objectInfo.CreateAt
+				payloadSize := objectInfo.PayloadSize
+				if objectInfo.IsUpdating {
+					shadowObject, found := k.GetShadowObjectInfo(ctx, bucket.BucketName, objectInfo.ObjectName)
+					if !found {
+						result = errors.New("shadow object not found")
+						ctx.Logger().Error("shadow object not found", "bucket", bucket.BucketName, "object", objectInfo.ObjectName)
+						continue Exit
+					}
+					priceTime = shadowObject.UpdatedAt
+					payloadSize = shadowObject.PayloadSize
+				}
+
+				lockAmount, _, err := k.GetObjectLockFee(ctx, priceTime, payloadSize)
+				if err != nil {
+					result = errors.New("get object lock fee failed")
+					ctx.Logger().Error("get object lock fee failed", "bucket", bucket.BucketName, "object", objectInfo.ObjectName, "error", err)
+					continue Exit
+				}
+				expectedLockBalance = expectedLockBalance.Add(lockAmount)
+			}
+		}
+
+		if expectedLockBalance.IsPositive() {
+			_, ok := lockBalanceMap[bucket.PaymentAddress]
+			if !ok {
+				lockBalanceMap[bucket.PaymentAddress] = sdkmath.ZeroInt()
+				lockBalanceDetailMap[bucket.PaymentAddress] = []Detail{}
+			}
+			lockBalanceMap[bucket.PaymentAddress] = lockBalanceMap[bucket.PaymentAddress].Add(expectedLockBalance)
+			lockBalanceDetailMap[bucket.PaymentAddress] = append(lockBalanceDetailMap[bucket.PaymentAddress],
+				Detail{bucket.BucketName, expectedLockBalance})
+		}
+	}
+
+	if result != nil { // if already has error, do not check the following
+		ctx.Logger().Info("stop checking payment data due to error")
+		return result
+	}
+
+	// compare lock balance: expected -> actual side
+	for address, expectedLockBalance := range lockBalanceMap {
+		streamRecord, found := streamRecordMap[address]
+		if !found {
+			result = errors.New("comparing lock balance - stream record not found")
+			ctx.Logger().Error("comparing lock balance - stream record not found", "address", address)
```

Comment:
Do you think we should panic if there is anyting wrong?

Reply:
It will panic when calling this function.
Here, it does not panic so that we can continue the following codes to print all errors at once.